### ISSUE TITLE
[filter-effects] SVG filter for invert not working properly

### ISSUE THREAD

**Issue by blakegearin** [w3c/fxtf-drafts#462]
Hello, I noticed that the [SVG filter for `invert`](https://www.w3.org/TR/filter-effects-1/#invertEquivalent) does not appear to be working properly.

```html
<feComponentTransfer>
  <feFuncR type="table" tableValues="[amount] (1 - [amount])"/>
  <feFuncG type="table" tableValues="[amount] (1 - [amount])"/>
  <feFuncB type="table" tableValues="[amount] (1 - [amount])"/>
</feComponentTransfer>
```

An amount of `0.5` appears over-lighten. Unless I'm mistaken that an `amount` of `0.5` is equivalent to a CSS filter's `50%`...?

I found that using `feFuncA` instead of `feFuncR`, `feFuncG`, and `feFuncB` appears to work correctly.

```html
<feComponentTransfer>
  <feFuncA type="table" tableValues="[amount] (1 - [amount])"/>
</feComponentTransfer>
```

Comparison:

```html
<p>Original Color</p>
<div style=""></div>
<p>CSS filter invert</p>
<div class="base"></div>
<p>SVG filter invert on W3</p>
<div class="base" style="filter: url(#w3-invert)"></div>
<p>SVG filter invert with feFuncR</p>
<div class="base" style="filter: url(#feFuncR-invert)"></div>
<svg>
  <filter id="w3-invert">
    <feComponentTransfer>
      <feFuncR type="table" tableValues=".5 .5"/>
      <feFuncG type="table" tableValues=".5 .5"/>
      <feFuncB type="table" tableValues=".5 .5"/>
    </feComponentTransfer>
</filter>
</svg>
<svg>
  <filter id="feFuncR-invert">
    <feComponentTransfer>
      <feFuncA type="table" tableValues=".5 .5"/>
    </feComponentTransfer>
  </filter>
</svg>

<style>
.base {
  height: 100px; 
  width: 100px; 
  background-color: #000000;  
}
</style>
```

Image preview:

<img width="228" alt="image" src="https://user-images.githubusercontent.com/17950836/180668956-abb7e76c-8388-4bd8-88e2-81374ac57f88.png">

**Comment by blakegearin** [comment]
As a side note, it would be nice if this spec was consistently using `[1 - amount]` or `(1 - [amount])` instead of both variations.

**Comment by blakegearin** [comment]
It appears you must specify sRGB to resolve this issue: `<filter color-interpolation-filters="sRGB">...</filter>`

**Comment by BigBadaboom** [comment]
Yes, that's correct.  The SVG spec says that `<filter>` elements default to
the linear colorspace. Whereas the CSS spec specifies that filter functions
always operate in the sRGB colour space.

References:

https://www.w3.org/TR/SVG11/single-page.html#painting-ColorInterpolationFiltersProperty

color-interpolation-filters has an initial value of linearRGB



https://drafts.fxtf.org/filter-effects/#FilterProperty

color-interpolation-filters has no affect for Filter Functions. Filter
> Functions must operate in the sRGB color space.

**Comment by svgeesus** [comment]
> An amount of 0.5 on black appears over-lightened (approximately #bcbcbc instead of #808080). 

Which is why filter operations are carried out in linear-light sRGB, which is defined here:

https://drafts.csswg.org/css-color-4/#predefined-sRGB-linear

> Whereas the CSS spec specifies that filter functions always operate in the sRGB colour space.

Which is dumb, and will need to be fixed at some point.

**Comment by svgeesus** [comment]
That wording was [added in 2017(https://github.com/w3c/fxtf-drafts/commit/1e2a527ca8e69bbbbd7f6a5e80bd08eac01529b5).

The few tests for invert on WPT are kind of vague, eg [this one](https://github.com/web-platform-tests/wpt/blob/master/css/filter-effects/backdrop-filters-invert.html) whose [reference](https://github.com/web-platform-tests/wpt/blob/master/css/filter-effects/backdrop-filters-invert-ref.html) **also** uses the `invert()` function!!

**Comment by mfreed7** [comment]
> That wording was [added in 2017](https://github.com/w3c/fxtf-drafts/commit/1e2a527ca8e69bbbbd7f6a5e80bd08eac01529b5).
> 
> The few tests for invert on WPT are kind of vague, eg [this one](https://github.com/web-platform-tests/wpt/blob/master/css/filter-effects/backdrop-filters-invert.html) whose [reference](https://github.com/web-platform-tests/wpt/blob/master/css/filter-effects/backdrop-filters-invert-ref.html) **also** uses the `invert()` function!!

In my defense, that test compares `backdrop-filter` to `filter` to make sure they do the same thing. Using other filter functions such as `blur` for that causes flakiness due to sub-pixel stuff.
